tests/features/AddressTest.php

```diff
@@ -0,0 +1,245 @@
+
+        $this->assertEquals(
+            $this->testModel->street,
+            $this->testModel->fresh()->street
+        );
+    }
+
+    /** @test */
+    public function can_set_default_address()
+    {
+        $secondaryAddress = $this->createSecondaryAddress();
+
+        $this->patch(
+            route('core.addresses.setDefault', $secondaryAddress->id, false))
+            ->assertStatus(200);
+
+        $this->assertFalse(
+            $this->testModel->fresh()->is_default
+        );
+
+        $this->assertTrue(
+            $secondaryAddress->fresh()->is_default
+        );
+    }
+
+    /** @test */
+    public function can_set_default_a_default_address()
+    {
+        $this->assertTrue(
+            $this->testModel->fresh()->is_default
+        );
+
+        $this->patch(
+            route('core.addresses.setDefault', $this->testModel->id, false))
+            ->assertStatus(200);
+
+        $this->assertTrue(
+            $this->testModel->fresh()->is_default
+        );
+    }
+
+    /** @test */
+    public function can_get_edit_address()
+    {
+        $this->get(
+            route('core.addresses.edit', $this->testModel->id, false),
+            $this->testModel->toArray())
+            ->assertStatus(200)
+            ->assertJsonStructure(['form' => 'form']);
+    }
+
+    /** @test */
+    public function can_get_create_address()
+    {
+        $this->get(
+            route('core.addresses.create', $this->testModel->id, false),
+            $this->testModel->toArray())
+            ->assertStatus(200)
+            ->assertJsonStructure(['form' => 'form']);
+    }
+
+    /** @test */
+    public function can_delete_address()
+    {
+        $this->assertNotNull($this->testModel);
+
+        $this->delete(route('core.addresses.destroy', $this->testModel->id, false))
+            ->assertStatus(200);
+
+        $this->assertNull($this->testModel->fresh());
+    }
+
+    /** @test */
+    public function can_get_label_attribute()
+    {
+        $this->testModel->street = "street";
+        $this->testModel->number = "number";
+        $this->testModel->city = "city";
+        $country = $this->testModel->country;
+        $country->name = "country";
+
+        $this->assertEquals(
+            'number street, city, country',
+            $this->testModel->getLabelAttribute()
+        );
+    }
+
+    /** @test */
+    public function cannot_delete_a_default_address_while_having_secondary_address()
+    {
+        $secondaryAddress = $this->createSecondaryAddress();
+
+        $this->delete(route('core.addresses.destroy', $this->testModel->id, false))
+            ->assertStatus(555)
+            ->assertJsonStructure(["message"]);
+
+        $this->assertNotNull($this->testModel->fresh());
+        
+        $this->assertNotNull($secondaryAddress->fresh());
+    }
+
+    /**
+     * @test
+     */
+    public function cannot_delete_a_addressable_while_having_restrict_address()
```

Comment:
it's either `can_delete_an_addressable` or `can_delete_addressable`. For the sake of shorter names, I'd go with the second :)